android/app/src/main/java/net/minetest/minetest/GameActivity.java

```diff
@@ -85,6 +85,17 @@ protected void onResume() {
 		makeFullScreen();
 	}
 
+	private native void saveSettings();
+
+	@Override
+	protected void onStop() {
+		super.onStop();
+		// Avoid losing setting changes in case the app is onDestroy()ed later.
+		// Saving stuff in onStop() is recommended in the Android activity
+		// lifecycle documentation.
+		saveSettings();
+	}
```

Comment:
not that I think it need changing but couldn't we poll Stopped from irrlicht instead of adding a new java method?

Reply:
I'm wasn't sure whether polling would be reliable enough - could you somehow get the native code to terminate before it notices that the app is stopped? perhaps with enough client lag? (probably not)

also, a callback is cleaner IMO, and you don't need separate implementations for the mainmenu and the game

Reply:
Alright, should be fine.